testing/integration/upgrade_test.go

```diff
@@ -936,18 +940,22 @@ func TestUpgradeBrokenPackageVersion(t *testing.T) {
 	versionList, err := aac.GetVersions(ctx)
 	require.NoError(t, err)
 	require.NotEmpty(t, versionList.Versions, "Artifact API returned no versions")
-	latestVersion := versionList.Versions[len(versionList.Versions)-1]
 
-	t.Logf("Upgrading to version %q", latestVersion)
+	// transform and reverse the version list and find the most recent version that is different from the broken version returned
+
+	// look for 1 version with the current major that is lower than the current one
```

Comment:
I'm guessing it's 1 thing? AM I right?

```suggestion
	// transform and reverse the version list and find the most recent version that is different from the broken version returned
	// look for 1 version with the current major that is lower than the current one
```